crates/swc_ecma_transforms_base/src/resolver/mod.rs

```diff
@@ -344,35 +344,17 @@ impl<'a> Resolver<'a> {
 
         if self.in_type {
             self.current.declared_types.insert(ident.sym.clone());
-            let mark = self.current.mark;
-
-            ident.span = if mark == Mark::root() {
-                ident.span
-            } else {
-                let span = ident.span.apply_mark(mark);
-                if cfg!(debug_assertions) && LOG {
-                    debug!("\t-> {:?}", span.ctxt());
-                }
-                span
-            };
-            return;
+        } else {
+            self.current
+                .declared_symbols
+                .insert(ident.sym.clone(), kind);
         }
 
         let mark = self.current.mark;
 
-        self.current
```

Comment:
Removal of this logic seems dangerous to me at that time, but now I'm not sure why I feel it dangerous.